### PR TITLE
replace some props on the object returned by getMode during compile time

### PR DESCRIPTION
**Pull request by erwinmombay**
getMode().localDev == false
getMode().test == false
getMode().minified == true

### CODE REVIEW

File: src/error.js [hunk 1]
```diff
@@ -107,8 +107,7 @@ function reportErrorToServer(message, filename, line, col, error) {
   if (this && this.document) {
     makeBodyVisible(this.document);
   }
-  const mode = getMode();
```
Comment (erwinmombay) [inline src/error.js]: @cramforce PTAL. works but unfortunately I don't detect re-aliasing right now, is there a way to get  the type information of a return type? (I'm sure there is I just don't know how to get to it)

Comment (cramforce) [inline src/error.js]: Lets make this illegal.

File: build-system/runner/src/org/ampproject/AmpPass.java [hunk 2]
```diff
@@ -56,56 +60,99 @@ public AmpPass(AbstractCompiler compiler, Set<String> stripTypeSuffixes) {
...
-      String name = expression.getQualifiedName();
-      if (name == null) {
-        return false;
-      }
+  /**
+   * Predicate for any <code>fnQualifiedName</code>.<code>props</code> call.
+   * example:
+   *   isFunctionInvokeAndPropAccess(n, "getMode", "test"); // matches `getMode().test`
+   *
+   * @param n
```
Comment (cramforce) [inline build-system/runner/src/org/ampproject/AmpPass.java]: Remove these

File: build-system/runner/test/org/ampproject/AmpPassTest.java [hunk 3]
```diff
@@ -17,7 +17,7 @@
       "dev.fine");
 
   @Override protected CompilerPass getProcessor(Compiler compiler) {
-    return new AmpPass(compiler, suffixTypes);
+    return new AmpPass(compiler, true, suffixTypes);
```
Comment (cramforce) [inline build-system/runner/test/org/ampproject/AmpPassTest.java]: Add comment for that is `true`.

**Comment by cramforce** [comment]
Looks great. Lets add a presubmit check or compile error for `getMode()` aliasing?

**Comment by erwinmombay** [comment]
@cramforce will do

File: src/error.js (hunk 1 above)
Comment (erwinmombay) [inline src/error.js]: done.

File: build-system/runner/test/org/ampproject/AmpPassTest.java (hunk 3 above)
Comment (erwinmombay) [inline build-system/runner/test/org/ampproject/AmpPassTest.java]: sorry, did you mean to do something like `return new AmpPass(compiler, /* isProd */ true, suffixTypes);` ?

Comment (erwinmombay) [inline build-system/runner/test/org/ampproject/AmpPassTest.java]: done.

**Comment by erwinmombay** [comment]
@cramforce ready for review again. PTAL

**Comment by cramforce** [comment]
LGTM

**Pull request merged**